### ISSUE TITLE
Upgrade 1.1.8 -> 2.0.7 fails

### ISSUE THREAD

**Issue by sjoerdie** [open-formulieren/open-forms#3117]
### Product versie / Product version

1.1.8

### Omschrijf het probleem / Describe the bug

Gemeente Venray:
1.1.8 to 2.0.1

```
  Apply all migrations: accounts, admin, admin_index, analytics_tools, appointments, auth, authtoken, axes, bag, brp, config, contenttypes, cookie_consent, cspreports, custom_field_types, digid_eherkenning_oidc_generics, django_camunda, django_yubin, emails, forms, jcc, kvk, logging, microsoft, mozilla_django_oidc_db, multidomain, of_authentication, otp_static, otp_totp, payments, payments_ogone, prefill, prefill_haalcentraal, products, qmatic, registrations_email, registrations_microsoft_graph, registrations_objects_api, sessions, simple_certmanager, stuf, stuf_bg, stuf_zds, submissions, timeline_logger, two_factor, upgrades, zgw_apis, zgw_consumers
Running migrations:
Traceback (most recent call last):
  File "src/manage.py", line 24, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/app/src/openforms/forms/migrations/0046_convert_advanced_logic.py", line 13, in convert_frontend_advanced_logic
    advanced_formio_logic_to_backend_logic(form_definition)
  File "/app/src/openforms/forms/utils.py", line 408, in advanced_formio_logic_to_backend_logic
    parsed_trigger = parse_trigger(
  File "/app/src/openforms/forms/utils.py", line 321, in parse_trigger
    and configuration_wrapper[trigger_component_key]["type"] == "selectboxes"
KeyError: 'heeftUEenPartner'
  Applying forms.0046_convert_advanced_logic...+ export PGHOST=/var/run/postgresql
```
1.1.8 to 2.0.7

```
  2023-05-25 12:32:08,349 WARNING openforms.forms.utils  Logic trigger with non-existing component (heeftUEenPartner)
  Applying forms.0046_convert_advanced_logic... OK
Traceback (most recent call last):
  File "src/manage.py", line 24, in <module>
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 419, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.8/site-packages/django/core/management/__init__.py", line 413, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 354, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 398, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/base.py", line 89, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.8/site-packages/django/core/management/commands/migrate.py", line 244, in handle
    post_migrate_state = executor.migrate(
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/executor.py", line 227, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/migration.py", line 126, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.8/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/app/src/openforms/forms/migrations/0047_fix_broken_converted_rules.py", line 13, in fix_broken_converted_rules
    fix_broken_rules(rules)
  File "/app/src/openforms/forms/utils.py", line 456, in fix_broken_rules
    is_valid = serializer.is_valid()
  File "/usr/local/lib/python3.8/site-packages/rest_framework/serializers.py", line 748, in is_valid
    self._validated_data = self.run_validation(self.initial_data)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/serializers.py", line 618, in run_validation
    value = self.to_internal_value(data)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/serializers.py", line 666, in to_internal_value
    validated = self.child.run_validation(item)
  File "/usr/local/lib/python3.8/site-packages/rest_framework/serializers.py", line 429, in run_validation
    value = self.validate(value)
  File "/app/src/openforms/forms/api/serializers/logic/action_serializers.py", line 149, in validate
    data["form_step_uuid"] = match.kwargs["uuid"]
KeyError: 'uuid'
```

### Screen resolution

None

### Device

None

### OS

None

**Comment by sergei-maertens** [comment]
There were some upgrade fixes in 1.1.9 - can you try updating to 1.1.10 first and then to 2.0.7?

See https://open-forms.readthedocs.io/en/stable/changelog.html#id11

**Comment by joeribekker** [comment]
Refinement: Should be fixed in 2.0.x. the oneline fix was to add `.replace('/api/v1', '/api/v2')` in some utils function used in migration :)
